extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -475,39 +475,27 @@ export class Bind {
         this.addMacros_(),
         this.addBindingsForNodes_([root]),
       ]);
-    }).then(results => {
-      dev().info(TAG, '⤷', 'Δ:', results);
+    }).then(() => {
       // Listen for DOM updates (e.g. template render) to rescan for bindings.
       root.addEventListener(AmpEvents.DOM_UPDATE, e => this.onDomUpdate_(e));
       // In dev mode, check default values against initial expression results.
       if (getMode().development) {
         return this.evaluate_().then(results => this.verify_(results));
       }
-    }).then(() => this.checkReadiness_(root));
-  }
-
-  /**
-   * Bind is "ready" when its initialization completes _and_ all <amp-state>
-   * elements' local data is parsed and processed (not remote data).
-   * @param {!Node} root
-   * @private
-   */
-  checkReadiness_(root) {
-    const ampStates = root.querySelectorAll('AMP-STATE');
-    if (ampStates.length > 0) {
-      const whenBuilt = toArray(ampStates).map(el => el.whenBuilt());
-      Promise.all(whenBuilt).then(() => this.onReady_());
-    } else {
-      this.onReady_();
-    }
-  }
-
-  /**
-   * @private
-   */
-  onReady_() {
-    this.viewer_.sendMessage('bindReady', undefined);
-    this.dispatchEventForTesting_(BindEvents.INITIALIZE);
+    }).then(() => {
+      const ampStates = root.querySelectorAll('AMP-STATE');
+      // Force all query-able <amp-state> elements to parse local data instead
+      // of waiting for runtime to build them all.
+      const whenParsed = toArray(ampStates).map(el => {
+        return el.getImpl().then(impl => impl.parseAndUpdate());
```

Comment:
I'm not sure the elements are guaranteed to be upgraded to custom elements yet. You'll need to [wait for upgrade](https://github.com/ampproject/amphtml/blob/bf504ab750f64b9343c228a1756d38d95bbba7c8/src/dom.js#L758-L773) first.